extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -839,40 +839,45 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /**
-   * Render creative via SafeFrame.
-   * @param {!ArrayBuffer} creativeBody  The creative, as raw bytes.
+   * Render the creative via some "cross domain iframe that accepts the creative
+   * in the name attribute".  This could be SafeFrame or the AMP-native
+   * NameFrame.
+   *
+   * @param {!ArrayBuffer} creativeBody
    * @return {!Promise} awaiting load event for ad frame
    * @private
    */
-  renderViaSafeFrame_(creativeBody) {
-    this.emitLifecycleEvent('renderSafeFrameStart');
-    return utf8Decode(creativeBody).then(creative => {
-      /** @const {!Element} */
-      const iframe = createElementWithAttributes(
-          /** @type {!Document} */(this.element.ownerDocument),
-          'iframe', Object.assign({
-            'height': this.element.getAttribute('height'),
-            'width': this.element.getAttribute('width'),
-            'src': SAFEFRAME_IMPL_PATH + '?n=0',
-            'name': `${SAFEFRAME_VERSION};${creative.length};${creative}`,
-          }, SHARED_IFRAME_PROPERTIES));
-      return this.iframeRenderHelper_(iframe);
-    });
-  }
-
-  renderViaNameFrame_(creativeBody) {
+  renderViaNameAttrOfXDomIframe_(creativeBody) {
+    const method = this.nonAmpCreativeRenderMethod_;
+    dev().assert(method == CROSS_ORIGIN_RENDERING_MODE.SAFEFRAME ||
+        method == CROSS_ORIGIN_RENDERING_MODE.NAMEFRAME,
+        'Unrecognized A4A cross-domain rendering mode: %s', method);
     this.emitLifecycleEvent('renderSafeFrameStart');
     return utf8Decode(creativeBody).then(creative => {
+      let srcPath;
+      let nameData;
+      if (method == CROSS_ORIGIN_RENDERING_MODE.SAFEFRAME) {
```

Comment:
Oh, right.  The `if ... else if` is a leftover.  Switch makes more sense now.

Re: location of the `assert`.  I think it should be outside.  It's dev mode so that we can see it if we manage to get here by mistake, but it will throw out of the code altogether.  If we should happen to get here in prod mode, I'd like a gentler return.